homestar-invocation/src/receipt.rs

```diff
@@ -178,3 +186,61 @@ impl TryFrom<Receipt<Ipld>> for Pointer {
         Ok(Pointer::new(receipt.to_cid()?))
     }
 }
+
+impl<T> JsonSchema for Receipt<T> {
+    fn schema_name() -> String {
+        "receipt".to_owned()
+    }
+
+    fn schema_id() -> Cow<'static, str> {
+        Cow::Borrowed("homestar-invocation::receipt::Receipt")
+    }
+
+    fn json_schema(gen: &mut SchemaGenerator) -> Schema {
+        let meta_schema = SchemaObject {
+            instance_type: Some(SingleOrVec::Single(InstanceType::Object.into())),
+            metadata: Some(Box::new(Metadata {
+                title: Some("Receipt metadata".to_string()),
+                description: Some(
+                    "Receipt metadata including the operation that produced the receipt"
+                        .to_string(),
+                ),
+                ..Default::default()
+            })),
+            object: Some(Box::new(ObjectValidation {
+                properties: BTreeMap::from([("op".to_owned(), <String>::json_schema(gen))]),
+                required: BTreeSet::from(["op".to_string()]),
+                ..Default::default()
+            })),
+            ..Default::default()
+        };
+
+        let schema = SchemaObject {
+            instance_type: Some(SingleOrVec::Single(InstanceType::Object.into())),
+            metadata: Some(Box::new(Metadata {
+                title: Some("Receipt".to_string()),
+                description: Some("A computed receipt".to_string()),
+                ..Default::default()
+            })),
+            object: Some(Box::new(ObjectValidation {
+                properties: BTreeMap::from([
+                    ("ran".to_owned(), gen.subschema_for::<Pointer>()),
```

Comment:
we can probably do to_string/to_owned once on these fields, instead of twice (in required and properties).

Reply:
We could, but we would need to clone anyways because the first use would take ownership.

Reply:
Yeah. I think there's some room for optimizations here, but no biggie.